components/Budget/ViewBudgetModal.jsx

```diff
@@ -0,0 +1,65 @@
+import { useState, useEffect, useContext } from "react";
+import { View, Text, ScrollView } from "react-native";
+import { ref, onValue, get } from "firebase/database";
+import { AlertContext } from "../Alert/AlertProvider";
+import { firebase_db } from "../../firebase.config";
+import BudgetInfo from "./BudgetInfo.jsx";
+import LoadingWheel from "../Loading/LoadingWheel.jsx";
+
+export default function ViewBudgetModal({ productionCode }) {
+    const { setAlert } = useContext(AlertContext);
+    const [budgets, setBudgets] = useState([]);
+    const [loading, setLoading] = useState(true);
+    const db = firebase_db;
+
+    useEffect(() => {
```

Comment:
Main Error: app is crashing when clicking view budget on production EI0PB9, maybe try joining the production and seeing this?

Reply:
think its when there's a budget with no participants

Reply:
I should probably be filtering by whether the user is a member of the budget tho tbf

Reply:
done that now too